src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -17,7 +24,49 @@ public LineService(LineRepository lineRepository) {
     }
 
     public LineResponse saveLine(LineRequest request) {
+        throwExceptionExistsDuplicateName(request);
         Line persistLine = lineRepository.save(request.toLine());
         return LineResponse.of(persistLine);
     }
+
+    private void throwExceptionExistsDuplicateName(LineRequest request) {
+        Optional<Line> savedLine = lineRepository.findByName(request.getName());
+        if (savedLine.isPresent()) {
+            throw new DuplicateLineNameException("중복된 노선명이 있습니다.");
+        }
+    }
+
+    public List<LineResponse> getLines() {
+        return lineRepository.findAll()
+                .stream()
+                .map(LineResponse::of)
+                .collect(Collectors.toList());
+    }
+
+    public LineResponse getLine(Long id) {
+        return lineRepository.findById(id)
+                .map(LineResponse::of)
+                .orElseThrow(throwLineNotFoundException(id));
+    }
+
+    public LineResponse modifyLine(Long id, LineRequest lineRequest) {
+        return lineRepository.findById(id)
+                .map(line -> line.update(lineRequest.toLine()))
+                .map(LineResponse::of)
+                .orElseThrow(throwLineNotFoundException(id));
+    }
+
+    private Supplier<LineNotFoundException> throwLineNotFoundException(Long id) {
+        return () -> new LineNotFoundException(String.format("아이디 %d는 없는 노선입니다.", id));
+    }
+
+    public void deleteLine(Long id) {
+        throwLineNotFoundExceptionIfHasNotLine(id);
+        lineRepository.deleteById(id);
```

Comment:
inline해도 괜찮지 않을까요 ? 🤔 
```suggestion
        Line savedLine = lineRepository.findById(id)
                .orElseThrow(throwLineNotFoundException(id));
        lineRepository.delete(savedLine);
```

Reply:
네 굳이 메소드로 뺄 필요가 없었을 것 같아요~
메소드명 읽는게 더 가독성이 안좋아 보이는 것 같습니다.